test/parse/bad-references.txt

```diff
@@ -0,0 +1,29 @@
+;;; TOOL: wat2wasm
+;;; ARGS: --enable-function-references
+;;; ERROR: 1
+(module
+  ;; Implicit type declarations are not allowed
+  (type $t0 (func (param (ref 2))))
+  (type $t1 (func (param (ref 33)) (result (ref 444))))
+
+  (func $f (param $p1 (ref 5555))
+    (local $p2 (ref 66666))
+  )
+)
+(;; STDERR ;;;
+out/test/parse/bad-references.txt:6:31: error: reference type out of range: 2 (max: 2)
```

Comment:
The error message seems to suggest that 2 should be within range?

Reply:
I just copied an existing error template:
```
(module
  (func $f (type 0))
)
```

I get:
```
test.wast:8:18: error: function type variable out of range: 0 (max 0)
  (func $f (type 0))
```

What should we use instead? `(< 0)` or `(less than 0)`

Reply:
Fair enough.  LGTM as is then.